src/systemcmds/tests/test_parameters.cpp

```diff
@@ -157,14 +178,326 @@ bool ParameterTest::ResetAllExcludesWildcard()
+
+	// load saved params
+	if (param_load_default() != PX4_OK) {
+		PX4_ERR("param_save_default failed");
+		ret = true;
+	}
+
+	// check every param
+	for (auto p : test_params) {
+		if (param_type(p) == PARAM_TYPE_INT32) {
+
+			int32_t get_val = 0.0f;
+
+			if (param_get(p, &get_val) != PX4_OK) {
+				PX4_ERR("param_get failed for: %d", p);
+				ut_assert("param_set_no_notification failed", false);
+			}
+
+			ut_compare("value for param doesn't match default value", p, get_val);
+		}
+
+		if (param_type(p) == PARAM_TYPE_FLOAT) {
+			float get_val = 0.0f;
+
+			if (param_get(p, &get_val) != PX4_OK) {
+				PX4_ERR("param_get failed for: %d", p);
+				ut_assert("param_set_no_notification failed", false);
+			}
+
+			ut_compare_float("value for param doesn't match default value", p, (float)p + MAGIC_FLOAT_VAL, 0.001f);
+		}
+	}
+
+	return ret;
+}
+
+bool ParameterTest::exportImportAll()
+{
+	static constexpr float MAGIC_FLOAT_VAL = 0.217828f;
+
+	// backup current parameters
+	const char *param_file_name = PX4_ROOTFSDIR "/fs/microsd/param_backup";
+	int fd = open(param_file_name, O_WRONLY | O_CREAT, PX4_O_MODE_666);
+
+	if (fd < 0) {
+		PX4_ERR("open '%s' failed (%i)", param_file_name, errno);
+		return false;
+	}
+
+	int result = param_export(fd, false);
+
+	if (result != PX4_OK) {
+		PX4_ERR("param_export failed");
+		close(fd);
+		return false;
+	}
+
+	close(fd);
+
+
+	bool ret = true;
+
+	int N = param_count();
+
+	// set all params to corresponding param_t value
+	for (unsigned i = 0; i < N; i++) {
+
+		param_t p = param_for_index(i);
+
+		if (p == PARAM_INVALID) {
+			PX4_ERR("param invalid: %d(%d)", p, i);
+			break;
+		}
+
+		if (param_type(p) == PARAM_TYPE_INT32) {
+			const int32_t set_val = p;
+
+			if (param_set_no_notification(p, &set_val) != PX4_OK) {
+				PX4_ERR("param_set_no_notification failed for: %d", p);
+				ut_assert("param_set_no_notification failed", false);
+			}
+
+			int32_t get_val = 0;
+
+			if (param_get(p, &get_val) != PX4_OK) {
+				PX4_ERR("param_get failed for: %d", p);
+				ut_assert("param_set_no_notification failed", false);
+			}
+
+			ut_compare("value for param doesn't match default value", p, get_val);
+		}
+
+		if (param_type(p) == PARAM_TYPE_FLOAT) {
+			const float set_val = (float)p + MAGIC_FLOAT_VAL;
+
+			if (param_set_no_notification(p, &set_val) != PX4_OK) {
+				PX4_ERR("param_set_no_notification failed for: %d", p);
+				ut_assert("param_set_no_notification failed", false);
+			}
+
+			float get_val = 0.0f;
+
+			if (param_get(p, &get_val) != PX4_OK) {
+				PX4_ERR("param_get failed for: %d", p);
+				ut_assert("param_set_no_notification failed", false);
+			}
+
+			ut_compare("value for param doesn't match default value", p, (float)p + MAGIC_FLOAT_VAL);
+		}
+	}
+
+	// save
+	if (param_save_default() != PX4_OK) {
+		PX4_ERR("param_save_default failed");
+		return false;
+	}
+
+	// zero all params and verify, but don't save
+	param_control_autosave(false);
+
+	for (param_t p = 0; p < N; p++) {
```

Comment:
`param_t p = param_for_index(i);` should be used here as well (and one more further down).

Reply:
done